### PR TITLE
Fix outdated sections of the c++ style guide

### PR DESCRIPTION
**Pull request by tibetiroka**
In the recent years, the [C++ Style Guide](https://endless-sky.github.io/styleguide/styleguide.xml) has become quite outdated. In this issue, I'd like to list (and fix) several of these problems.

Feel free to change the wording and presentation of the new sections. I'm probably not the best person for this kind of writing job, but it seems like nobody really cares anyway. 

I'm also bumping the revision number from 1.0 to 1.1 to account for these changes.

## Indentation in blank lines
Apparently, this was changed recently to disallow any indentation on blank lines, and we even have a CI that checks this.

## Function calls
This section is included only to account for the mismatch of formatting in the source and test files. In the source, no parentheses have spaces inside, whereas in the unit tests it has become the standard. According to quyykk, this behaviour is not to be changed.

## Opening braces
Same as the previous section; tests put the braces on the same line for readability.

# Other issues
There are some other issues that are not addressed in this PR, but I'd still like to have mentioned.

## File extensions
There are also various .hpp files in the project for specific use cases. Maybe this is worth mentioning, since the guide forbids anything other than .h / .cpp

## Line length
The current hard wrap is at 120 characters. There are currently over a hundred lines in the source code that exceed this limit; I already have a branch with the fixes for the main repo, but you might want to consider increasing this limit nevertheless.

### CODE REVIEW

File: styleguide/styleguide.xml [hunk 1]
```diff
@@ -647,6 +646,34 @@ x = temp;
 </BODY>
 </STYLEPOINT>
 
+<STYLEPOINT title="Function calls">
+<SUMMARY>Do not put a space inside the parentheses for method or function calls.</SUMMARY>
```
Comment (quyykk) [inline styleguide/styleguide.xml:650]: This is exactly the same as the previous point.

Comment (tibetiroka) [reply styleguide/styleguide.xml:650]: I may be reading that the wrong way but the previous point talked more about spaces between the name and the following parenthesis. Could be reworded to match this one though.

Comment (quyykk) [reply styleguide/styleguide.xml:650]: Oh yeah I'm dumb

Comment (tibetiroka) [reply styleguide/styleguide.xml]: dang it, I fixed it 5 seconds ago

Comment (tibetiroka) [reply styleguide/styleguide.xml]: Is the new version good or should I roll it back?

Comment (quyykk) [reply styleguide/styleguide.xml]: it's fine

File: styleguide/styleguide.xml [hunk 2]
```diff
@@ -851,6 +875,28 @@ if(theThing != done)
...
+        WHEN( "a fine is levied" ) {
+            REQUIRE( account.TotalDebt() == 0 );
+            account.AddFine(10000);
+            THEN ( "debt is incurred" ) {
+                REQUIRE( account.TotalDebt() == 10000 );
+            }
+        }
+    }
+}
+</CODE_SNIPPET>
```
Comment (quyykk) [inline styleguide/styleguide.xml:899]: Can you shorten both code snippets a bit so that it only shows the current rule being discussed?

**Review by quyykk — APPROVED** [review]

**Pull request merged**